cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetTypeService.java

```diff
@@ -101,6 +131,39 @@ public DatasetTypeService(DatasetTypeManager typeManager, NamespaceQueryAdmin na
     this.authenticationContext = authenticationContext;
     this.cConf = cConf;
     this.impersonator = impersonator;
+    this.txClientService = txClientService;
+    this.datasetFramework = datasetFramework;
+    this.txExecutorFactory = txExecutorFactory;
+    Map<String, String> emptyArgs = Collections.emptyMap();
+    this.datasetCache = new MultiThreadDatasetCache(
+      new SystemDatasetInstantiator(datasetFramework, null, null), txClientService, NamespaceId.SYSTEM, emptyArgs, null,
```

Comment:
yeah, everything just moved from ``DatasetTypeManager``. Its needed because we want to do enforcement even on system datasets, so we will grant privileges on these guys to the user who creates them. That way, we have less code bypassing authorization.